opm/simulators/wells/SingleWellState.cpp

```diff
@@ -262,14 +262,20 @@ update_producer_targets(const Well& ecl_well, const SummaryState& st)
     case Well::ProducerCMode::GRUP:
     case Well::ProducerCMode::THP:
     case Well::ProducerCMode::BHP:
+        // Setting all rates to zero, they will be initialized properly in
+        // a call to WellInterface::initializeProducerWellState() later, which will
+        // use the reservoir state to find a better initial value.
+        // This also applies to the ORAT/WRAT/GRAT above, but then only the
+        // rate not set in the above will be modified in
+        // WellInterface::initializeProducerWellState().
         if (this->pu.phase_used[BlackoilPhases::Liquid]) {
```

Comment:
I might miss something here, and not talking about the approach either, but from the code level, will we miss something by `std::fill(this->surface_rates.begin(), this->surface_rates.end(), 0.);` here?